src/test_utils/public/enzyme_helpers.tsx

```diff
@@ -17,22 +17,21 @@
  * under the License.
  */
 
-/**
- * Components using the react-intl module require access to the intl context.
- * This is not available when mounting single components in Enzyme.
- * These helper functions aim to address that and wrap a valid,
- * intl context around them.
- */
-
-import React from 'react';
-import { I18nProvider, intlShape } from '@kbn/i18n/react';
-import { mount, shallow, render } from 'enzyme';
+import { I18nProvider, InjectedIntl, intlShape } from '@kbn/i18n/react';
+import { mount, ReactWrapper, render, shallow } from 'enzyme';
+import React, { ReactElement } from 'react';
+import { IntlProvider } from 'react-intl';
 
 // Use fake component to extract `intl` property to use in tests.
-const { intl } = mount(<I18nProvider><br /></I18nProvider>)
-  .find('IntlProvider').instance().getChildContext();
+const { intl } = (mount(
+  <I18nProvider>
+    <br />
+  </I18nProvider>
+).find('IntlProvider') as ReactWrapper<{}, {}, IntlProvider>)
```

Comment:
Why is this better than importing at the top of the file? Typescript will take care of removing the import if it's just for types right? Why do we want to stress that we import only the type?

Reply:
> Why is this better than importing at the top of the file? Typescript will take care of removing the import if it's just for types right?

I _believe_ it will.

> Why do we want to stress that we import only the type?

Well, I know it's arguable, but in this case it's purely psychological thing - I don't want any code to import from `react-intl` directly, ever. Especially `InltProvider` since `@kbn/i18n` exports its own replacement for `IntlProvider` - `I18nProvider`, so my point was to limit usage of this "leak" in this specific place and just for type declaration so that no one ever accidentally grep it and think that importing it is appropriate.

Am I paranoiac? :)